### DISCUSSION TITLE
Tried 8.0, failed, scurried back to 7.7 😂 Help?

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#718] [App and O/S Config Questions] [answered]
whoever said, "If it ain't broke..." obviously never met me.

Couldn't resist trying out v8 but was nervous about messing up my 7.7 install, which has been humming along nicely every day... kudos!

Here's what I did - almost certainly a process error on my part 🤷‍♂️ - let me know where I might have gone wrong.

1.  Stopped all services (`pm2 stop all`) and took a backup (`tar -czvf njspc ./nodejs-poolController`) - same for dash, REM
- this backup is what saved me when I couldn't get v8 up and running on the first try, i just bumped node back down to v14, restored and njsPC 7.7 was humming along again in no time

2. back to the v8 attempt - I bumped node up to latest v18 (`18.16.0`)

for each service:
3. `git pull`
4. `pm2 stop [service]`
5. `npm i`
6. `pm2 start [service]`

1st attempt, `git pull` screamed about conflict in `package-lock.json` so i moved that out of the app dir and then tried again.  Then `git pull` was successful for each service.

Then I started looking at the logs.  Probably should have started each service one at a time, but here's what I observed for each service.

### njsPC
service never seemed to start before it would restart,  didn't even see an error in the logs, it just cycled the following:
```
> nodejs-poolcontroller@8.0.1 start
> npm run build && node dist/app.js


> nodejs-poolcontroller@8.0.1 build
> tsc


> nodejs-poolcontroller@8.0.1 start
> npm run build && node dist/app.js


> nodejs-poolcontroller@8.0.1 build
> tsc


> nodejs-poolcontroller@8.0.1 start
> npm run build && node dist/app.js


> nodejs-poolcontroller@8.0.1 build
> tsc
```

### dash & REM

 having trouble tracking down the console output when trying to start up v8 of both service, but both of these got further than njsPC.  But then pm2 watchdog detected an update and kept restarting services - ex. something like `detected change in dist/boards/Constants.js.map, restarting`

I used the same pm2 ecosystem file from the wiki that defines what gets watched and never had an issue with v7.  Do I need to change something in my ecosystem file?

```
module.exports = {
  apps : [
    {
      "name": "REM",
      "script": "npm",
      "args": [
        "start"
      ],
      "cwd": "/home/pi/relayEquipmentManager",
      "restart_delay": 10000,
      "watch": [
        "boards",
        "config",
        "connections",
        "devices",
        "gpio",
        "i2c-bus",
        "logger",
        "pages",
        "pinouts",
        "spi-adc",
        "web",
        "package.json"
      ],
      "watch_delay": 5000,
      "kill_timeout": 15000
    },
    {
      "name": "dashPanel",
      "script": "npm",
      "args": [
        "start"
      ],
      "cwd": "/home/pi/nodejs-poolController-dashPanel",
      "restart_delay": 10000,
      "watch": [
        "pages",
        "scripts",
        "server",
        "package.json"
      ],
      "watch_delay": 5000,
      "kill_timeout": 15000
    },
    {
      "name": "njsPC",
      "script": "npm",
      "args": [
        "start"
      ],
      "cwd": "/home/pi/nodejs-poolController",
      "restart_delay": 10000,
      "watch": [
        "config",
        "controller",
        "logger",
        "web",
        "package.json"
      ],
      "watch_delay": 5000,
      "kill_timeout": 15000
    }
  ]
};

```

Sorry to be a pain - just want to join you chaps on the latest and greatest!

**Comment by rstrouse** [comment]
You likely need to upgrade your node version.  If you are using nvm type the following command `nvm install 18`

**Reply by johnny2678** [reply]
> You likely need to upgrade your node version. If you are using nvm type the following command `nvm install 18`

`node -v` was reporting `v18.16.0` prior to running `npm i`

Does it need to be a different version?

**Reply by rstrouse** [reply, marked as answer]
Nope 18.16.0 should work fine.  What error are you getting?  Try starting njsPC by running the following commands.
`pm2 stop njsPC`
`cd ~/nodejs-PoolController`
`npm start`

**Reply by rstrouse** [reply]
Oh and one other thing.  It probably wasn't `screaming` per se about package-lock.json.  It more likely was simply converting it to the version required by your install and like everything out there these days over-ventilating about what it needed to do.

**Reply by johnny2678** [reply]
here's the `git pull` error:
```
pi@PiPoolv2:~/nodejs-poolController $ git pull
remote: Enumerating objects: 301, done.
remote: Counting objects: 100% (300/300), done.
remote: Compressing objects: 100% (126/126), done.
remote: Total 301 (delta 185), reused 273 (delta 171), pack-reused 1
Receiving objects: 100% (301/301), 520.00 KiB | 3.08 MiB/s, done.
Resolving deltas: 100% (185/185), completed with 34 local objects.
From https://github.com/tagyoureit/nodejs-poolController
   4ed27b05..5aa14551  master     -> origin/master
   0fefae97..8033c226  8.0.0      -> origin/8.0.0
 * [new tag]           v8.0.0     -> v8.0.0
 * [new tag]           v8.0.1     -> v8.0.1
Updating 4ed27b05..5aa14551
error: Your local changes to the following files would be overwritten by merge:
	package-lock.json
Please commit your changes or stash them before you merge.
Aborting
```
can't recall any scenario where I would have manually edited the `package-lock.json` - and it's saying it on all 3 services.

to get around I have to move `package-lock.json` out of the folder then 'git pull' again

**Reply by rstrouse** [reply]
You should perform a `git reset --hard` and that will clear this issue.

**Comment by johnny2678** [comment]
ok!

I tried again this morning and it's working now!  Only thing I did differently was start each service manually with `npm start`, instead of using pm2.  Seems like things need to start and settle on the first run before employing pm2 and it's watchdog.

Now, all 3 services are running fine with pm2 and all of my nodered http/socket calls are working too!  You guys are the best.

**Comment by johnny2678** [comment]
qq - if I already have an rs485 connection, there's no added benefit in the ScreenLogic integration, correct?

**Reply by rstrouse** [reply]
There is no advantage.  RS485 will always be more performant and you in fact cannot have both.
